src/tools/crawler/main.rs

```diff
@@ -75,33 +75,68 @@ fn start_logger(default_level: LevelFilter) {
         .init();
 }
 
+/// Parses and converts `String` values found in a `Vec` to valid `SocketAddr`.
+///
+/// # Input
+///
+/// Valid inputs can be in the following forms:
+/// - IP + port (both IPv4 and IPv6 are valid)
+/// - IP (Default `ZCASH_P2P_DEFAULT_PORT` will be appended)
+/// - DNS + port
+/// - DNS
+fn parse_addrs(seed_addrs: Vec<String>, crawler: &Crawler) -> Vec<SocketAddr> {
+    let mut parsed_addrs = Vec::with_capacity(seed_addrs.len());
+
+    for seed_addr in seed_addrs {
+        // First, try parsing as a `SocketAddr`.
+        if let Ok(addr) = seed_addr.parse::<SocketAddr>() {
+            parsed_addrs.push(addr);
+            continue;
+        }
+        // User may supply an IP address without a port,
+        // append `ZCASH_P2P_DEFAULT_PORT` in that case.
+        if let Ok(addr) = seed_addr.parse::<IpAddr>() {
+            parsed_addrs.push(SocketAddr::new(addr, ZCASH_P2P_DEFAULT_PORT));
+            info!(parent: crawler.node().span(), "no port specified for address: {}, using default: {}", seed_addr, ZCASH_P2P_DEFAULT_PORT);
+            continue;
+        }
+        // If above failed, try to do a DNS lookup instead.
+        //
+        // We make sure to remove remove the port if it exists.
```

Comment:
🤔
maybe there will be out there some DNS addresses with a non-default port and we won't allow the user to crawl those addresses

we can keep it simple - append a port if it is not provided

Reply:
Port is extracted and added later, when hostname is resolved to IPaddr - so the port is preserved as the user specified.

Reply:
aaa, I see! That's why some unit test input examples would be nice to have here - it would quickly clarify something like this